oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/collections/IterableUtils.java

```diff
@@ -212,4 +214,46 @@ public static <T> T[] toArray(final Iterable<T> itr, final Class<T> type) {
         final Collection<T> collection = itr instanceof Collection ? (Collection<T>) itr : ListUtils.toList(itr);
         return collection.toArray(t);
     }
+
+    /**
+     * Splits an Iterable into an Iterator of sub-iterators, each of the specified size.
+     *
+     * @param <T> the type of elements in the itr
+     * @param itr the itr to split, may not be null
+     * @param size the size of each sub-iterator, must be greater than 0
+     * @return an iterator of sub-iterators, each of the specified size
+     * @throws NullPointerException if the itr is null
+     * @throws IllegalArgumentException if size is less than or equal to 0
+     */
+    public static <T> Iterable<List<T>> partition(final Iterable<T> itr, final int size) {
+        if (itr == null) {
```

Comment:
Why can't we use "requireNonNull" here? (or even do not do it at all?)

Reply:
IMO, we should fail fast.

Reply:
addressed in https://github.com/apache/jackrabbit-oak/pull/2111/commits/8103c0116fc53d00436f65a5a0f2339b6c9b3ff0

Reply:
Not having the null check would fail fast as well here.

My opinion is that requireNonNull is completely unnecessary noise in 99% of the cases where it's used.